Vector/Model/Room/MXRoom+Vector.m

```diff
@@ -0,0 +1,372 @@
+/*
+ Copyright 2015 OpenMarket Ltd
+ 
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ 
+ http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MXRoom+Vector.h"
+
+#import "AvatarGenerator.h"
+
+@interface MXRoom ()
```

Comment:
Are you sure that we can define class extension here (whereas this extension is already defined in MXRoom.m)?
We may have a discussion about this point. Another solution to handle the added observers is to use associated objects.